### PR TITLE
New labeling system - using colors and prefixes

### PR DESCRIPTION
**Pull request by jsignell**
ALTERNATIVE TO #16 

Ok I was looking at https://github.com/cli/cli/labels and was feeling inspired. I took a crack at a new labeling system that is much less verbose and takes advantage of alphabetical sorting and colors. I also changed the settings on the action so that the pre-existing labels with matching names will have their colors updated and descriptions updated, but all un-listed labels will not be removed. 

Other influences for these labels are: the categories on the changelog: https://docs.dask.org/en/stable/changelog.html and the pre-existing labels on dask and distributed.

### CODE REVIEW

File: .github/labels.yml
Comment (jacobtomlinson) [inline .github/labels.yml]:
```suggestion
# prefixed with "good"
```

**Review by jacobtomlinson — APPROVED** [review]
This is much better, thanks @jsignell.

Thinking about the `needs foo` labels, I could imagine a `needs review` would also be useful. I'm also unsure who `needs attention` requires attention from, is it the user, the maintainer, etc?

**Comment by jsignell** [comment]
Yeah "needs attention" is kind of like stale, so I think it's ambiguous who it needs attention from. I could see going down a a different path where we make these: "needs user input" and "needs maintainer input"

File: .github/labels.yml
Comment (fjetter) [inline .github/labels.yml:53]: What is core?

Comment (jsignell) [reply .github/labels.yml:53]: That's an existing label in dask https://github.com/dask/dask/labels/core but I'm not sure if it overlaps too much with the new labels (like maybe `hygiene`)

Comment (jrbourbeau) [reply .github/labels.yml:53]: I believe this is label was for low-ish level issues that impact a large part of the `dask/dask` codebase (e.g. graph optimization is a topic which applies to arrays, dataframes, bags, and delayed). Though I agree "core" is a vague term. I don't have strong feelings on keeping it or dropping it. We can always iterate later.

**Review by jrbourbeau — APPROVED** [review]
Thanks @jsignell. I like the readability of these labels more than the ones over in https://github.com/dask/.github/pull/16. Happy to give this a try and iterate as needed

**Comment by jsignell** [comment]
ok! I am going to merge this tomorrow morning unless there is more feedback

**Pull request merged**
